### DISCUSSION TITLE
Scanner .next() && .nextLine()

### DISCUSSION THREAD

**Discussion by mmasias** [mmasias/23-24-prg1#17] [Announcements]
¿Alguien podría decir la diferencia (o diferencias) entre los métodos .next() y .nextLine() de la clase Scanner?

**Comment by mmasias** [comment]
¿Nadie?

**Reply by Alejandrojuarez0105** [reply]
next() no lee espacio
nextLine() si los lee

**Reply by mmasias** [reply]
|||
|-|-|
Delimitadores|.next(): Este método lee la entrada hasta que encuentra un delimitador, que por defecto es un espacio en blanco. Así, .next() solo leerá una palabra a la vez.
||.nextLine(): Este método lee la entrada hasta que encuentra una nueva línea (\n). Por lo tanto, .nextLine() puede leer una línea completa de texto, incluyendo espacios.
Comportamiento con la nueva línea|.next(): Ignora cualquier espacio en blanco y espera hasta que encuentra texto no espaciado para comenzar a leer.
||.nextLine(): Lee todo, incluyendo espacios en blanco y termina de leer cuando encuentra una nueva línea. También capturará la nueva línea si es el primer carácter.
Uso de memoria|.next(): Al leer menos datos a la vez, .next() puede ser más eficiente en términos de memoria en comparación con .nextLine().
||.nextLine(): Como puede leer líneas completas de texto, podría utilizar más memoria si la línea es muy larga.
Aplicación|.next(): Es útil cuando necesitas analizar la entrada palabra por palabra o cuando la entrada está delimitada por espacios o algún otro delimitador.
||.nextLine(): Es útil cuando quieres leer líneas completas de texto.

Estas diferencias hacen que cada método sea más adecuado para diferentes situaciones. 

- Por ejemplo, si estás leyendo entrada del usuario palabra por palabra, .next() podría ser la elección correcta. 
- Por otro lado, si estás leyendo líneas completas de texto, .nextLine() sería más apropiado.
